src/GraphQL/Wrappers/Response/ViolationResponse.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\graphql\GraphQL\Wrappers\Response;
+
+use Drupal\graphql\GraphQL\Wrappers\Violation\ViolationCollection;
+
+/**
+ * Base class for responses containing the violations.
```

Comment:
So this response means: the request was a valid graphql request but there could be validation violations that the client needs to check. If there are violations then the mutation was not successful. So the status code 200 is a bit misleading if there are violations, because 200 means success?

Reply:
That `code` property is confusing, I agree, and we should definitely agree on whether we need it. And if yes, how to use it.

But let me at least explain why that `code` property was added. It was introduced on the `ResponseInterface` level, because we used it in `RedirectResponse`. As there can be multiple redirect response codes, we need to know which one the frontend should set.

For violations we just used default 200, because that's what normal non-critical responses do. If you login via regular login form and your password is wrong, Drupal also responds with 200. But this gives us actually no benefit at all and seems redundant.

Could be used maybe as some kind of violation code, so frontend knows more about the problem (besides violations messages). But honestly I am really not sure what to do with this `code` property :)

Reply:
Yes this code is pretty much ignored outside of a Redirect response. For redirects it kind of makes sense to know if a Redirect is permanent or not. But other than that.not really sure how it can be used. Do we need it in violations? Maybe it doesn't have to be part of `ResponseInterface` but can be specific to responses that need it.

Reply:
@joaogarin removed code property. I still thing it might be useful, because even correct response may have eg `201` code. But as it's redundant at the moment I removed it from response interface and also from violation response.

Reply:
I am ok with that :+1: will be pretty exciting if we are able to roll this out to the module itself. will make our lives easier but also will allow us to share a lot more.